### PR TITLE
updating normal for loops to range based for loops 

### PR DESCRIPTION
**Pull request by Sameeer12**
we are porting to modern cpp

Signed-off-by: Sameer Gupta <guptasameerprem@gmail.com>

_Please use the following template to help us managing pull requests._

## Summary of Changes

_Describe what your pull request changes to CGAL (this can be skipped if it solves an issue already in the tracker or if it is a Feature or Small Feature submitted to the CGAL Wiki)._

## Release Management

* Affected package(s):
* Issue(s) solved (if any): fix #0000, fix #0000,...
* Feature/Small Feature (if any):
* Link to compiled documentation (obligatory for small feature) [*wrong link name to be changed*](httpssss://wrong_URL_to_be_changed/Manual/Pkg)
* License and copyright ownership:

### CODE REVIEW

File: Triangulation_2/doc/Triangulation_2/CGAL/Triangulation_vertex_base_with_info_2.h [hunk 1]
```diff
@@ -15,10 +15,10 @@ which provides the `Point_2`. It is strongly
 recommended to instantiate this parameter 
 with the traits class used for the triangulation. 
 This ensures that the point type defined by `Triangulation_vertex_base_with_info_2` matches the point type defined by 
-the triangulation. 
+the triangulation.
```
Comment (r1walz) [inline Triangulation_2/doc/Triangulation_2/CGAL/Triangulation_vertex_base_with_info_2.h]: @Sameeer12 Your changes should not affect whitespaces unless required. Please revert them back to original.

File: Triangulation_2/include/CGAL/Constrained_Delaunay_triangulation_2.h [hunk 2]
```diff
@@ -690,8 +687,8 @@ void
 Constrained_Delaunay_triangulation_2<Gt,Tds,Itag>::
 flip_around(List_vertices& new_vertices)
 {
-  typename List_vertices::iterator itv=new_vertices.begin();
-  for( ; itv != new_vertices.end(); itv++) {
+ // typename List_vertices::iterator itv=new_vertices.begin();
+  for( auto itv : new_vertices){
```
Comment (r1walz) [inline Triangulation_2/include/CGAL/Constrained_Delaunay_triangulation_2.h]: Leave a space _after_ `for` **not** after `(`. Or follow the [coding conventions](https://doc.cgal.org/latest/Manual/devman_code_format.html).

File: Triangulation_2/include/CGAL/Constrained_Delaunay_triangulation_2.h [hunk 3]
```diff
@@ -690,8 +687,8 @@ void
 Constrained_Delaunay_triangulation_2<Gt,Tds,Itag>::
 flip_around(List_vertices& new_vertices)
 {
-  typename List_vertices::iterator itv=new_vertices.begin();
-  for( ; itv != new_vertices.end(); itv++) {
+ // typename List_vertices::iterator itv=new_vertices.begin();
```
Comment (r1walz) [inline Triangulation_2/include/CGAL/Constrained_Delaunay_triangulation_2.h]: Why comment this out? Simply, remove it completely.

File: Triangulation_2/include/CGAL/Constrained_triangulation_plus_2.h [hunk 4]
```diff
@@ -110,8 +110,8 @@ Default::Get< Tr_, Constrained_Delaunay_triangulation_2<
     void
     write_faces(OutputIterator out)
     {
-      for(typename std::vector<TFace>::reverse_iterator 
-            it = faces.rbegin(); it != faces.rend(); ++it) { 
+      for(auto it : faces)
```
Comment (r1walz) [inline Triangulation_2/include/CGAL/Constrained_triangulation_plus_2.h]: It is a reverse iterator, keep that in mind. @nk-nikunj can suggest what to do in this case.

File: Triangulation_2/include/CGAL/draw_triangulation_2.h [hunk 5]
```diff
@@ -35,7 +35,7 @@ struct DefaultColorFunctorT2
   }
 };
 
-// Viewer class for T2 
+// Viewer class for T2
```
Comment (r1walz) [inline Triangulation_2/include/CGAL/draw_triangulation_2.h]: Again, your editor removed that trailing space. I know it's not required, but it should not go into this commit at least.

File: Triangulation_2/include/CGAL/Triangulation_2.h [hunk 6]
```diff
@@ -668,7 +668,7 @@ std::ptrdiff_t insert(InputIterator first, InputIterator last,
 
     Vertex_handle v_hint;
     Face_handle hint;
-    for (std::size_t i = 0; i < points.size(); ++i ) {
+    for (auto& i : boost::counting_range(0,points.size) {
```
Comment (NK-Nikunj) [inline Triangulation_2/include/CGAL/Triangulation_2.h]: forgot a bracket in the end

File: Triangulation_2/include/CGAL/Constrained_triangulation_plus_2.h (hunk 4 above)
Comment (Sameeer12) [reply Triangulation_2/include/CGAL/Constrained_triangulation_plus_2.h]: Thanks for pointing that point. I will amend the change.

**Pull request closed without merging**